distributed-job-manager/src/main/java/ru/fix/distributed/job/manager/strategy/EvenlySpreadAssignmentStrategy.java

```diff
@@ -0,0 +1,69 @@
+package ru.fix.distributed.job.manager.strategy;
+
+import ru.fix.distributed.job.manager.model.AssignmentState;
+import ru.fix.distributed.job.manager.model.JobId;
+import ru.fix.distributed.job.manager.model.WorkItem;
+import ru.fix.distributed.job.manager.model.WorkerItem;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+public class EvenlySpreadAssignmentStrategy implements AssignmentStrategy {
+
+    @Override
+    public AssignmentState reassignAndBalance(
+            AssignmentState availability,
+            AssignmentState prevAssignment,
+            AssignmentState currentAssignment,
+            Map<JobId, List<WorkItem>> itemsToAssign
+    ) {
+        int workersCount = currentAssignment.size();
+        if (workersCount == 0) {
+            return currentAssignment;
+        }
+
+        int itemsToAssignSize = itemsToAssign.values().stream()
+                .mapToInt(List::size)
+                .sum();
+        int canBeTakenFromPreviousPerWorker = itemsToAssignSize / workersCount;
+
+        for (Map.Entry<WorkerItem, List<WorkItem>> worker : prevAssignment.entrySet()) {
+            int itemsAddedFromPrevious = 0;
+
+            for (WorkItem workItem : worker.getValue()) {
+                if (!currentAssignment.containsKey(worker.getKey())) {
+                   continue;
+                }
+                if (itemsAddedFromPrevious >= canBeTakenFromPreviousPerWorker) {
+                    break;
+                }
+                currentAssignment.addWorkItem(worker.getKey(), workItem);
```

Comment:
could you please  clarify and add the case when Job1 no longer can be launched on worker2. You are treating prevAssignment as if all itmes in it continue to stay where they are, but actually availability could change and prevAssignment is no longer a correct state!